include/zephyr/sched_server/cbs.h

```diff
@@ -0,0 +1,162 @@
+/*
+ *  Copyright (c) 2024 Instituto Superior de Engenharia do Porto (ISEP)
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ *
+ * @brief Constant Bandwidth Server (CBS) public API
+ */
+
+#ifndef ZEPHYR_CBS
+#define ZEPHYR_CBS
+
+#include <zephyr/kernel.h>
+#include <zephyr/sched_server/internal/cbs_internal.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef CONFIG_CBS
+
+/**
+ * @defgroup cbs_apis Constant Bandwidth Server (CBS) APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/**
+ * @brief CBS job format.
+ *
+ * A job pushed to a CBS is a regular function that must
+ * have a void pointer as an argument (can be NULL if not
+ * needed) and return nothing. jobs are pushed to the CBS by
+ * invoking k_cbs_push_job().
+ */
+typedef void (*cbs_callback_t)(void *arg);
+
+/**
+ * @cond INTERNAL_HIDDEN
+ */
+
+typedef struct {
+	cbs_callback_t function;
+	void *arg;
+} cbs_job_t;
```

Comment:
No typedef struct names in zephyr please, absent special circumstances like by-value opaque types.  Should just be "struct cbs_job"

Reply:
ok, I've removed typedef structs.